pkg/controllers/proxy/network_services_controller.go

```diff
@@ -140,12 +141,17 @@ func (ln *linuxNetworking) ipAddrAdd(iface netlink.Link, ip string, addRoute boo
 		return nil
 	}
 
-	// TODO: netlink.RouteReplace which is replacement for below command is not working as expected. Call succeeds but
-	// route is not replaced. For now do it with command.
```

Comment:
I believe this comment by @murali-reddy doesn’t hold true anymore. (Happy to be proven wrong.)

What I did to verify that `netlink.RouteReplace` works as intended:

1. Exec into a `kube-router` pod and install go and git:

   ```
   apk add go git
   ```

2. Create a file called `test-netlink.go`:

	```go
   	package main

	import (
		"github.com/golang/glog"
		"github.com/vishvananda/netlink"
		"golang.org/x/sys/unix"
		"net"
	)

	func main() {
		err := netlink.RouteReplace(&netlink.Route{
			Dst:       &net.IPNet{IP: net.ParseIP("5.5.5.5"), Mask: net.IPv4Mask(255, 255, 255, 255)},
			LinkIndex: 1,
			Table:     unix.RT_TABLE_LOCAL,
			Protocol:  unix.RTPROT_KERNEL,
			Scope:     unix.RT_SCOPE_HOST,
			Type:      unix.RTN_LOCAL,
			Src:       net.ParseIP("127.0.0.1"),
		})
		if err != nil {
			glog.Errorf("Failed to replace route to service. Error: %v", err)
		}
	}
	```

3. Install dependencies and pin `netlink` to the version which `kube-router` uses:

   ```sh
   go get
   cd $(go env GOPATH)/src/github.com/vishvananda/netlink
   git checkout f504738125a57f35f87fc30fb69b8df75237ccde # Version from Gopkg.lock
   ```

4. Run `test-netlink.go` and verify that the route got added:

   ```
   cd -
   go run test-netlink.go
   ip route show table all | grep 5.5.5.5
   # Should print: local 5.5.5.5 dev lo table local proto kernel scope host src 127.0.0.1
   ```

5. Change the source IP address in `test-netlink.go`, re-run it, and verify that the route gets replaced correctly:

   ```sh
   sed -i 's/127.0.0.1/172.17.0.2/' test-netlink.go
   go run test-netlink.go
   ip route show table all | grep 5.5.5.5
   # Should print: local 5.5.5.5 dev lo table local proto kernel scope host src 172.17.0.2
   ```

Is this a correct way to verify that?